### ISSUE TITLE
Postgres needs access to scheduling information

### ISSUE THREAD

**Issue by btbonval** [medic/medic-webapp#1912]
Schedules could possibly change over time.

Within postgres, if we entirely recreate schedules (duplicating nools logic), then we might miss when parameters change _and_ we're exposing ourselves to loss of sync in logic.

For LG, we (or our partners) might hard code certain schedule parameters directly into analytics views in the short term.

In the long term, it'd be great if we had access to some kind of task trail, or nools debug, or something like this which preserves the logic performed in a way which can be queried and utilized accurately, post hoc. If a task was created using logic and parameters from 3 months ago, we'd like to have some way to know the result of that task as it was calculated 3 months ago.

**Comment by SCdF** [comment]
So I'm trying to understand this ticket. @garethbowen or @browndav is the gist that we need to modify how we use nools so it outputs results of certain events in a postgres-consumable fashion (ie couchdb documents), presumably things we'd be using to display Targets?

**Comment by browndav** [comment]
Yeah, sorry, it's not abundantly clear from the original description. Your intuition is correct: there's currently no persistent record of what tasks were presented to users, and the only persistent record of a task being completed is the side-effects of having completed it – e.g. a form submission. I _believe_ the goal here is to be able to answer questions about task completion rates and protocol adherence (e.g. How many tasks were completed more than two late between the periods of X and Y, and how late on average were they?).

IIRC, using Nools to generate SMS notifications (either as an alternative to tasks, or in addition) is also on the table, and would AFAICT require running the rules engine centrally in addition to on the mobile devices. IMO we should work with the product team to clarify the exact requirements before jumping in to this.
